File: AspNetMvcUnleashed/0672329980_source/Chapter20/VB/UnleashedBlog/UnleashedBlog/Models/EntityFramework/EntityFrameworkBlogRepository.vb
Public Class EntityFrameworkBlogRepository
    Inherits BlogRepositoryBase

    Private _entities As New BlogDBEntities()

    Private Function ConvertBlogEntryToBlogEntryEntity(ByVal entry As BlogEntry) As BlogEntryEntity
        Dim entity = New BlogEntryEntity()

        entity.Id = entry.Id
        entity.Author = entry.Author
        entity.Description = entry.Description
        entity.Name = entry.Name
        entity.DatePublished = entry.DatePublished
        entity.Text = entry.Text
        entity.Title = entry.Title
        Return entity
    End Function

    Protected Overrides Function QueryBlogEntries() As IQueryable(Of BlogEntry)
        Return From e In _entities.BlogEntryEntitySet _
               Select New BlogEntry With {.Id = e.Id, _
                                          .Author = e.Author, _
                                          .Description = e.Description, _
                                          .Name = e.Name, _
                                          .DateModified = e.DateModified, _
                                          .DatePublished = e.DatePublished, _
                                          .Text = e.Text, _
                                          .Title = e.Title}
    End Function


    Public Overrides Sub CreateBlogEntry(ByVal blogEntryToCreate As BlogEntry)
        Dim entity = ConvertBlogEntryToBlogEntryEntity(blogEntryToCreate)

        _entities.AddToBlogEntryEntitySet(entity)
        _entities.SaveChanges()
    End Sub

End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter17/CS/UnleashedBlog/UnleashedBlog/Models/EntityFramework/EntityFrameworkBlogRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace UnleashedBlog.Models.EntityFramework
{
    public class EntityFrameworkBlogRepository : BlogRepositoryBase
    {
        private BlogDBEntities _entities = new BlogDBEntities();

        private BlogEntryEntity ConvertBlogEntryToBlogEntryEntity(BlogEntry entry)
        {
            var entity = new BlogEntryEntity();

            entity.Id = entry.Id;
            entity.Author = entry.Author;
            entity.Description = entry.Description;
            entity.Name = entry.Name;
            entity.DatePublished = entry.DatePublished;
            entity.Text = entry.Text;
            entity.Title = entry.Title;
            return entity;
        }

        protected override IQueryable<BlogEntry> QueryBlogEntries()
        {
            return from e in _entities.BlogEntryEntitySet
                   select new BlogEntry 
                   { 
                        Id = e.Id,
                        Author = e.Author,
                        Description = e.Description,
                        Name = e.Name, 
                        DateModified = e.DateModified,
                        DatePublished = e.DatePublished,
                        Text = e.Text,
                        Title = e.Title
                   };
        }
        
        public override List<BlogEntry> ListBlogEntries()
        {
            return QueryBlogEntries().ToList();            
        }

        public override void CreateBlogEntry(BlogEntry blogEntryToCreate)
        {
            var entity = ConvertBlogEntryToBlogEntryEntity(blogEntryToCreate);

            _entities.AddToBlogEntryEntitySet(entity);
            _entities.SaveChanges();
        }

    }
}
